lib/membrane/video_compositor/queue/offline/element.ex

```diff
@@ -0,0 +1,289 @@
+defmodule Membrane.VideoCompositor.Queue.Offline.Element do
+  @moduledoc """
+  Module responsible for offline queueing strategy.
+
+  In this strategy frames are send to compositor only when all added input pads queues
+  with timestamp offset lower or equal to composed buffer pts
```

Comment:
Queue elements have multiple `"queues"`, each for a different input pad. Since some input pads have a nonzero timestamp offset, they shouldn't block the whole queue element on earlier buffers.
 
Imagine having `OfflineQueue` with output `fps` of 1 and 3 input pads:
- first with 0 timestamp offset,
- second with 1s timestamp offset,
- third with 2 timestamp offset.

This `OfflineQueue` should produce:
- first buffer with pts 0 only with first input pad frame
- second buffer with pts 1s with first and second input pads frames
- third buffer with pts 2s with first, second, and third input pads frames
- n-th with pts n-th  with first, second, and third input pads frames

Here I meant, the `OfflineQueue` element checking/composing the next buffer should only account for queues of those input pads, and that timestamps offsets are lower than or equal to the next buffer pts.

I guess calling those `"queues"` specific to Input Pad the same name as the whole element might be a little bit confusing here. Any ideas on how to rename it?

Reply:
I believe having `frames_and_events_queue` is slightly to long, while having `input_pad_queue` might be redundant (since we would refer to it as to `pad.input_pad_queue`). Neither me nor GPT could think of any other meaningful alternatives, so I would stick to the `queue` name ;)

Reply:
When it comes to the module doc, it's now perfectly clear to me (previously I have missed the "COMPOSED buffer" part)